src/codegen.rs

```diff
@@ -421,6 +425,27 @@ impl<'ctx> CodeGen<'ctx> {
         Ok((variable.name.to_string(), variable_type, initializer))
     }
 
+    /// generates assignment statements for initialized variables in the VAR-block
+    ///
+    /// # Arguments
+    /// * `blocks` - all declaration blocks of the current pou
+    fn generate_initialization_of_local_vars(
+        &self,
+        blocks : &Vec<VariableBlock>
+    )-> Result<(), String> {
+        let variables_with_initializers = blocks.iter()
+            .filter(|it| it.variable_block_type == VariableBlockType::Local)
```

Comment:
What do we do with inputs with defaults? are they even allowed as optional variables then? 
They can't fit here because this would override the parameter value.
I was thinking that the initialization could happen on the alloca call as well, but this would require us to save these values for later and use them when we find a function call, this makes it better.
Maybe we should, however, when filling input values also fill in default values.

Reply:
I dont believe that anything but VARs can be optional for functions. With FBs and Programs they can, because you can fill them via the instance:
```
PRG.x := 3; 
PRG();
``` 
 It would break implicit calls foo(1,2,3); But I'm not 100% sure

Reply:
You're right, I also don't think optionals can  be used here officially (Our own codegen supports them for now, but it's just because they are supported for FBs)e.
Let's just solve that issue if it comes up later.